File: Scripts/results/generated_code_results/Flutter/Entertainment/musicAPP2/PlaylistScreen.dart
import 'package:flutter/material.dart';
class PlaylistScreen extends StatefulWidget {
@override
_PlaylistScreenState createState() => _PlaylistScreenState();
}
class _PlaylistScreenState extends State<PlaylistScreen> {
final List<Map<String, String>> songs = [
{'title': 'Gully Boy', 'artist': 'Ranvir Sing'},
{'title': 'Apna Time Aayega', 'artist': 'Arijit Sing'},
{'title': 'Illegal Weapon 2.0', 'artist': 'Street Dancer 3D'},
{'title': 'Lagli Lahore Di', 'artist': 'Sreya Ghosal'},
{'title': 'Bezubaan Kab Se', 'artist': 'Kapal Dev'},
{'title': 'Nachi Nachi', 'artist': 'Sreya Ghoshal'},
];
int currentPlayingIndex = 2;
@override
Widget build(BuildContext context) {
return Scaffold(
body: Container(
decoration: BoxDecoration(
gradient: LinearGradient(
begin: Alignment.topCenter,
end: Alignment.bottomCenter,
colors: [Colors.blue, Colors.white],
),
),
child: SafeArea(
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Padding(
padding: const EdgeInsets.all(16.0),
child: Row(
mainAxisAlignment: MainAxisAlignment.spaceBetween,
children: [
Text(
'9:41',
style: TextStyle(color: Colors.white, fontSize: 16),
),
Row(
children: [
Icon(Icons.signal_cellular_alt, color: Colors.white),
SizedBox(width: 8),
Icon(Icons.wifi, color: Colors.white),
SizedBox(width: 8),
Icon(Icons.battery_full, color: Colors.white),
],
),
],
),
),
Padding(
padding: const EdgeInsets.all(16.0),
child: Row(
children: [
IconButton(
icon: Icon(Icons.arrow_back, color: Colors.white),
onPressed: () => Navigator.pop(context),
),
SizedBox(width: 8),
Expanded(
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text(
'Geri - Inder Chahal',
style: TextStyle(
color: Colors.white,
fontSize: 24,
fontWeight: FontWeight.bold,
),
),
Text(
'Pop Songs',
style: TextStyle(color: Colors.white70),
),
],
),
),
],
),
),
Padding(
padding: const EdgeInsets.all(16.0),
child: Row(
children: [
ClipRRect(
borderRadius: BorderRadius.circular(12),
child: Image.asset(
'assets/images/template.png',
width: 100,
height: 100,
fit: BoxFit.cover,
),
),
SizedBox(width: 16),
Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text(
'Songs',
style: TextStyle(color: Colors.white70),
),
Text(
'21 Songs',
style: TextStyle(
color: Colors.white,
fontSize: 18,
fontWeight: FontWeight.bold,
),
),
SizedBox(height: 8),
Text(
'Author',
style: TextStyle(color: Colors.white70),
),
Text(
'Various Artists',
style: TextStyle(
color: Colors.white,
fontSize: 18,
fontWeight: FontWeight.bold,
),
),
],
),
Spacer(),
Container(
decoration: BoxDecoration(
shape: BoxShape.circle,
color: Colors.pink,
),
child: IconButton(
icon: Icon(Icons.play_arrow, color: Colors.white),
onPressed: () {},
),
),
],
),
),
Expanded(
child: ListView.builder(
itemCount: songs.length,
itemBuilder: (context, index) {
final song = songs[index];
final isPlaying = index == currentPlayingIndex;
return ListTile(
leading: Text(
'${index + 1}'.padLeft(2, '0'),
style: TextStyle(
color: isPlaying ? Colors.blue : Colors.grey,
fontWeight: FontWeight.bold,
),
),
title: Text(
song['title']!,
style: TextStyle(
color: isPlaying ? Colors.blue : Colors.black,
fontWeight: FontWeight.bold,
),
),
subtitle: Text(song['artist']!),
trailing: IconButton(
icon: Icon(Icons.more_vert),
onPressed: () {},
),
onTap: () {
setState(() {
currentPlayingIndex = index;
});
Navigator.pushNamed(context, 'NowPlaying');
},
);
},
),
),
Container(
color: Colors.blue,
padding: EdgeInsets.symmetric(horizontal: 16, vertical: 8),
child: Row(
children: [
ClipRRect(
borderRadius: BorderRadius.circular(4),
child: Image.asset(
'assets/images/template.png',
width: 40,
height: 40,
fit: BoxFit.cover,
),
),
SizedBox(width: 12),
Expanded(
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text(
'Illegal Weapon 2.0',
style: TextStyle(color: Colors.white, fontWeight: FontWeight.bold),
),
Text(
'Street Dancer 3D',
style: TextStyle(color: Colors.white70),
),
],
),
),
IconButton(
icon: Icon(Icons.favorite_border, color: Colors.white),
onPressed: () {},
),
IconButton(
icon: Icon(Icons.pause, color: Colors.white),
onPressed: () {},
),
],
),
),
],
),
),
),
);
}
}

File: Scripts/results/generated_code_results/RN_gpt/Entertainment/musicAPP2/PlaylistScreen.js
import React from 'react';
import { View, Text, StyleSheet, Image, TouchableOpacity, ScrollView } from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';

const PlaylistScreen = ({ navigation }) => {
  return (
    <ScrollView style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.title}>Geri - Inder Chahal</Text>
        <Text style={styles.subtitle}>Pop Songs</Text>
        <Image source={require('../assets/snack-icon.png')} style={styles.guitarImage} />
        <View style={styles.info}>
          <Text style={styles.infoText}>Songs</Text>
          <Text style={styles.infoText}>21 Songs</Text>
          <Text style={styles.infoText}>Author</Text>
          <Text style={styles.infoText}>Various Artists</Text>
        </View>
        <TouchableOpacity style={styles.playButton}>
          <Icon name="play" size={24} color="#fff" />
        </TouchableOpacity>
      </View>

      <View style={styles.trackList}>
        {[
          { number: '01.', title: 'Gully Boy', artist: 'Ranvir Sing' },
          { number: '02.', title: 'Apna Time Aayega', artist: 'Arijit Sing' },
          { number: '03.', title: 'Illegal Weapon 2.0', artist: 'Street Dancer 3D', highlighted: true },
          { number: '04.', title: 'Lagli Lahore Di', artist: 'Sreya Ghosal' },
          { number: '05.', title: 'Bezuban Kab Se', artist: 'Kapal Dev' },
          { number: '06.', title: 'Nachi Nachi', artist: 'Sreya Ghoshal' },
        ].map((track, index) => (
          <TouchableOpacity key={index} style={styles.trackItem} onPress={() => navigation.navigate('NowPlaying')}>
            <Text style={[styles.trackNumber, track.highlighted && styles.highlighted]}>{track.number}</Text>
            <View>
              <Text style={[styles.trackTitle, track.highlighted && styles.highlighted]}>{track.title}</Text>
              <Text style={styles.trackArtist}>{track.artist}</Text>
            </View>
            <Icon name="ellipsis-h" size={24} color="#ccc" />
          </TouchableOpacity>
        ))}
      </View>

      <View style={styles.footer}>
        <Image source={require('../assets/snack-icon.png')} style={styles.footerImage} />
        <Text style={styles.footerTitle}>Illegal Weapon 2.0</Text>
        <Text style={styles.footerSubtitle}>Street Dancer 3D</Text>
        <View style={styles.footerActions}>
          <Icon name="heart" size={24} color="#ccc" />
          <TouchableOpacity onPress={() => {}}>
            <Icon name="pause" size={24} color="#fff" />
          </TouchableOpacity>
        </View>
      </View>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#F5F5F5',
  },
  header: {
    backgroundColor: '#3E82F7',
    padding: 20,
    borderBottomLeftRadius: 20,
    borderBottomRightRadius: 20,
  },
  title: {
    fontSize: 24,
    color: '#fff',
    fontWeight: 'bold',
  },
  subtitle: {
    fontSize: 16,
    color: '#fff',
    marginVertical: 5,
  },
  guitarImage: {
    width: 100,
    height: 100,
    borderRadius: 12,
    marginVertical: 10,
  },
  info: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  infoText: {
    color: '#fff',
    fontSize: 12,
  },
  playButton: {
    position: 'absolute',
    right: 20,
    bottom: -20,
    backgroundColor: '#FF4081',
    width: 50,
    height: 50,
    borderRadius: 25,
    justifyContent: 'center',
    alignItems: 'center',
  },
  trackList: {
    padding: 20,
  },
  trackItem: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginVertical: 10,
  },
  trackNumber: {
    fontSize: 16,
    color: '#9E9E9E',
    marginRight: 10,
  },
  trackTitle: {
    fontSize: 16,
    color: '#3C3C3C',
  },
  trackArtist: {
    fontSize: 12,
    color: '#9E9E9E',
  },
  highlighted: {
    color: '#3E82F7',
    fontWeight: 'bold',
  },
  footer: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#3E82F7',
    padding: 10,
    position: 'relative',
    borderTopLeftRadius: 20,
    borderTopRightRadius: 20,
  },
  footerImage: {
    width: 50,
    height: 50,
    borderRadius: 5,
  },
  footerTitle: {
    color: '#fff',
    fontSize: 14,
    fontWeight: 'bold',
    marginLeft: 10,
  },
  footerSubtitle: {
    color: '#fff',
    fontSize: 12,
    marginLeft: 10,
  },
  footerActions: {
    position: 'absolute',
    right: 10,
    flexDirection: 'row',
    alignItems: 'center',
  },
});

export default PlaylistScreen;